file.c

```diff
@@ -6529,6 +6529,633 @@ const char ruby_null_device[] =
  *  \Class \File extends module FileTest, supporting such singleton methods
  *  as <tt>File.exist?</tt>.
  *
+ *  === About the Examples
+ *
+ *  Many examples here use these variables:
+ *
+ *    # English text with newlines.
+ *    text = <<~EOT
+ *      First line
+ *      Second line
+ *
+ *      Fourth line
+ *      Fifth line
+ *    EOT
+ *
+ *    # Russian text.
+ *    russian = "\u{442 435 441 442}" # => "тест"
+ *
+ *    # Binary data.
+ *    data = "\u9990\u9991\u9992\u9993\u9994"
+ *
+ *    # Text file.
+ *    File.write('t.txt', text)
+ *
+ *    # File with Russian text.
+ *    File.write('t.rus', russian)
+ *
+ *    # File with binary data.
+ *    f = File.new('t.dat', 'wb:UTF-16')
+ *    f.write(data)
+ *    f.close
+ *
+ *  The examples also assume that \StringIO has been required:
+ *
+ *    require 'stringio'
+ *
+ *  == \File Access Modes
+ *
+ *  \Methods File.new and File.open each create a \File object for a given file path.
+ *
+ *  === \String Access Modes
+ *
+ *  \Methods File.new and File.open each may take string argument +mode+, which:
+ *
+ *  - Begins with a 1- or 2-character
+ *    {read/write mode}[rdoc-ref:File@Read-2FWrite+Mode].
+ *  - May also contain a 1-character {data mode}[rdoc-ref:File@Data+Mode].
+ *  - May also contain a 1-character
+ *    {file-create mode}[rdoc-ref:File@File-Create+Mode].
```

Comment:
I think only one of these two may exist, I think we should be clear about that here.

Reply:
f = File.new('t.xyz', 'wbx') # => #<File:t.xyz>